smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/endpoints/EndpointResolverGenerator.kt

```diff
@@ -62,4 +63,17 @@ class EndpointResolverGenerator(
         writer.write("")
         writer.write("extension DefaultEndpointResolver: EndpointResolver {}")
     }
+
+    private fun renderStaticResolver(writer: SwiftWriter) {
+        writer.write("")
+        writer.openBlock("public struct StaticEndpointResolver: EndpointResolver {", "}") {
```

Comment:
Can we provide a reusable static endpoint resolver?

The only difference between this endpoint resolver for each service is that the `resolve` method takes that client's `EndpointParams` so we can make that type generic in a reusable static endpoint resolver, then simply use a typedef in each client to provide a specialized endpoint resolver.